spec/facter/facts/solaris/processors/threads_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+describe Facts::Solaris::Processors::Threads do
+  describe '#call_the_resolver' do
+    subject(:fact) { Facts::Solaris::Processors::Threads.new }
+
+    let(:threads_per_core) { '4' }
```

Comment:
About string vs int here, the Solaris processors resolver (from what I understand) returns int values. We should confirm that this is the case, and mock the fact tests with the same types. We should always aim to mock with "real life" values, so we don't get confused in the future if/when we return to this code.

Can you do a quick pass through all the resolvers and cross-check if the facts are tested with the correct types for `threads_per_core` and `cores_per_socket`?